skimage/transform/registration.py

```diff
@@ -0,0 +1,131 @@
+from scipy import ndimage as ndi
+
+from .pyramids import pyramid_gaussian
+from ..measure import compare_mse
+
+__all__ = ['register_affine']
+
+
+def _parameter_vector_to_matrix(parameter_vector, N):
+    """
+    Converts the optimisation parameters to a 3x3 transformation matrix
+
+    The optimisation paramters are known as the parameter_vector and are composed of
+    the first 2 rows of the transformation matrix, as that is all that
+    is used in an affine transformation.
+
+    Parameters
+    ----------
+    parameter_vector : (N*(N+1)) array
+        Input array giving the argument of the minimum function to optimise against
+
+    Returns
+    -------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+    """
+
+    return np.concatenate(
+        (np.reshape(parameter_vector, (N, N + 1)), [[0] * N + [1]]), axis=0)
+
+
+def _matrix_to_parameter_vector(matrix):
+    """
+    Converts a 3x3 transformation matrix to the optimisation parameters
+
+    See the inverse function `_parameter_vector_to_matrix`.
+
+    Parameters
+    ----------
+    matrix : (N+1, N+1) array
+        A transformation matrix used to obtain a new image
+
+    Returns
+    -------
+    parameter_vector : (N*(N+1)) array
+        Output array giving the argument of the minimum function to optimise against
+
+    """
+
+    return matrix[:-1, :].ravel()
+
+
+def register_affine(reference, target, *, cost=compare_mse, nlevels=None,
+                    iter_callback=lambda img, matrix: None):
+    """
+    Returns a matrix which registers the target image to the reference image
+
+    Will only return an affine transformation matrix.
+
+    Parameters
+    ----------
+    reference : ndimage
+        A reference image to compare against the target
+
+    target : ndimage
+        Our target for registration. Transforming this image using the return value
+        `matrix` aligns it with the reference.
+
+    cost : function, optional
+        A cost function which takes two images and returns a score which is at a
+        minimum when images are aligned. Uses the mean square error as default.
+
+    nlevels : integer, optional
+        Change the maximum height we use for creating the Gaussian pyramid.
+        By default we take a guess based on the resolution of the image,
+        as extremely low resolution images may hinder registration.
+
+    iter_callback : function, optional
+        If given, this function is called once per pyramid level with the current
+        downsampled image and transformation matrix guess as the only arguments.
+        This is useful for debugging or for plotting intermediate results
+        during the iterative processs.
+
+    Returns
+    -------
+    matrix : array
+        A transformation matrix used to obtain a new image.
+        ndi.affine_transform(target, matrix) will align your target image.
+
+    Example
+    -------
+    >>> reference_image = camera()
```

Comment:
The TravisCI build is [currently failing](https://travis-ci.org/scikit-image/scikit-image/jobs/453856808) on the test for this docstring example:
```
092     Example
093     -------
094     >>> reference_image = camera()
UNEXPECTED EXCEPTION: NameError("name 'camera' is not defined",)
```

Adding this line to the example should do the trick:
```python
>>> from skimage.data import camera
```